packages/transactions/lib/transactions.js

```diff
@@ -0,0 +1,208 @@
+import {ApiRpc} from '../client/api-rpc';
+import uuid from 'uuid';
+import {getRealm} from '../core/realm';
+import {DOCK_TOKEN_UNIT} from '../core/format-utils';
+import {fetchTransactions} from '../core/subscan';
+import BigNumber from 'bignumber.js';
+import {Accounts} from './accounts';
+import { NetworkManager } from './network-manager';
+
+export const TransactionStatus = {
+  InProgress: 'pending',
+  Failed: 'failed',
+  Complete: 'complete',
+};
+
+/**
+ * Transactions Module
+ */
+export class Transactions {
+  constructor() {
+    this.transactions = [];
+  }
+
+  /**
+   * Get transactions module instance
+   * @returns module instance
+   */
+  static getInstance() {
+    if (!Transactions.instance) {
+      Transactions.instance = new Transactions();
+    }
+
+    return Transactions.instance;
+  }
+  
+  /**
+   * Load external transactions for the given account
+   * 
+   * @param {string} account 
+   */
+  async loadExternalTransactions(account) {
+    const realm = getRealm();
```

Comment:
validate `account`  before trying to do any work?

Reply:
fixed